docs/overrides.rst

```diff
@@ -0,0 +1,167 @@
+.. _`overrides`:
+
+=========
+Overrides
+=========
+
+Using Page Objects is a great way to isolate the extraction logic from the crawling logic.
+As a side effect, it is now pretty easy to **create a generic spider with a common crawling logic
+that works across different sites**. The unique missing requirement is to be able to
+configure different Page Objects for each different site, because the extraction logic
+surely changes from site to site.
+This is exactly the functionality that overrides provides.
+
+Let's see is with an example. The following is a a spider that extract books
+from `books.toscrape.com <http://books.toscrape.com/>`_ using Page Objects.
+
+.. code-block:: python
+
+    import scrapy
+    from web_poet import ItemWebPage, WebPage
+    from scrapy_poet import callback_for
+
+
+    class BookListPage(WebPage):
+
+        def book_urls(self):
+            return self.css('.image_container a::attr(href)').getall()
+
+
+    class BookPage(ItemWebPage):
+
+        def to_item(self):
+            return {
+                'url': self.url,
+                'name': self.css("title::text").get(),
+            }
+
+
+    class BooksSpider(scrapy.Spider):
+        name = 'books_spider'
+        start_urls = ['http://books.toscrape.com/']
+
+        def parse(self, response, page: BookListPage):
+            for url in page.book_urls():
+                yield response.follow(url, callback_for(BookPage))
+
+Note that the crawling logic is pretty simple and straightforward:
+
+1. Extract all books urls from the listing page
+2. For each book URL found in the step 1, fetch the page and extract the resultant item
+
+This logic should work without any change for different books sites because
+having pages with lists of books and then detail pages for the individual book is
+such a common way of structuring sites.
+
+Let's reuse the very same ``BooksSpider`` spider but now to extract books from a different
+page: `bookpage.com/reviews <https://bookpage.com/reviews>`_.
+
+We cannot reuse the Books to Scrape Page Objects in this case, so we have
+to implement new ones:
+
+.. code-block:: python
+
+    class BPBookListPage(WebPage):
```

Comment:
I'd expect it to be a subclass of `BookListPage`.

Thinking that the crawler is generic, it should work with a common base interface and extensions should inherit from it

Reply:
Inheriting from `BookListPage` have the problem that you would be inheriting implementation details that might not want to. But I understand your point, so consider then the example where you have a base PO abstract class and then you have a concrete implementation for each case. You can see this implementation in the example `books_04_overrides_02.py` here: https://github.com/scrapinghub/scrapy-poet/pull/43/files#diff-855ff677d82fe0f56107cdaeb283941ff477930235ce9da6e9c16fbe5758fdf8

It also makes a lot of sense to me. Nothing prevents the user to follow this approach. Maybe the question is if we should also follow this approach in the documentation.

Reply:
I think that'd be good to follow this approach in the tutorial, because having site-specific code in a generic spider makes the example less clear. What do you think about the following structure?

1. Present basic example, with a concrete, website-specific BookPages used in a spider - no changes needed  here, this is already in the tutorial.
2. Show how to refactor it to make the spider generic: empty default implementation, overrides for both websites.

Reply:
@kmike it makes sense to me. It will make tutorial longer, but also useful as will explain which is the way to organize multi-site spiders.

Reply:
@ejulio @kmike I've updated the documentation as @kmike suggested here https://github.com/scrapinghub/scrapy-poet/pull/43/commits/26277a86d4a43a4591e8d38ea6e5beaadbea9311

Reply:
Resolved for me.
Not sure about you @kmike